projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -85,13 +96,43 @@ export class CartesianLegend {
   }
 
   private drawLegendEntries(container: ContainerElement): void {
-    select(container)
-      .append('div')
-      .classed('legend-entries', true)
-      .selectAll('.legend-entry')
-      .data(this.series.filter(series => !series.hide))
-      .enter()
-      .each((_, index, elements) => this.drawLegendEntry(elements[index]));
+    const containerSelection = select(container);
+    if (!this.isGrouped) {
+      containerSelection
+        .append('div')
+        .classed('legend-entries', true)
+        .selectAll('.legend-entry')
+        .data(this.series.filter(series => !series.hide))
+        .enter()
+        .each((_, index, elements) => this.drawLegendEntry(elements[index]));
+    } else {
+      containerSelection
+        .selectAll('.legend-entries')
+        .data(this.groupedSeries)
+        .enter()
+        .append('div')
+        .attr('class', seriesGroup => `legend-entries group-${seriesGroup[0].assignedGroup!.id}`)
+        .each((seriesGroup, id, elems) => {
+          const legendEntriesSelection = select(elems[id]);
+
+          legendEntriesSelection
+            .selectAll('.legend-entries-title')
+            .data([seriesGroup])
+            .enter()
+            .append('div')
+            .classed('legend-entries-title', true)
+            .text(group => `${group[0].assignedGroup!.specName}:`)
```

Comment:
assignedGroup could be missing.

Reply:
I would recommend you keep it optional and just assign a default value if it's missing. It could be ''

Reply:
Done